### PR TITLE
Run every time all tests (if our code in hardhat fails)

### PR DESCRIPTION
**Pull request by gigimon**
Logs when our faucet not work and proxy restarting sometime
https://github.com/neonlabsorg/neon-compatibility/runs/4642641168?check_suite_focus=true#step:10:16

### CODE REVIEW

File: hardhat.config.js [hunk 1]
```diff
@@ -146,11 +173,11 @@ console.log(
 (async (addressFrom, addressTo) => {
   console.log(
     `address from = ${addressFrom} balance=`,
-    await web3.eth.getBalance(addressFrom)
+    await getBalance(addressFrom)
```
Comment (apetrovskiy) [inline hardhat.config.js:160]: This is only to display the balance increased by the call to faucet (and optionally via call to getBalance)

File: hardhat.config.js [hunk 2]
```diff
@@ -86,27 +86,54 @@ for (const f of fs.readdirSync(path.join(__dirname, 'hardhat'))) {
...
+      console.log(`Failed to send request to faucet: ${err.response.data}`);
+    }
+  }
+  if (success !== true){
+    console.log("Can't get a good response from faucet 10 times, please check faucet")
   }
 };
 
-const getBalance = async (address) => await web3.eth.getBalance(address);
+const getBalance = async (address) => {
```
Comment (apetrovskiy) [inline hardhat.config.js:114]: This function is only used in displaying account's balance, should we have here up to 10 times' call?

File: hardhat.config.js [hunk 3]
```diff
@@ -86,27 +86,54 @@ for (const f of fs.readdirSync(path.join(__dirname, 'hardhat'))) {
 const withOptimizations =
   argv.enableGasReport || argv.compileMode === 'production';
 
+
+if (Config.useFaucet){
+  console.log(`Use faucet: ${Config.faucetUrl}`)
```
Comment (apetrovskiy) [inline hardhat.config.js:91]: If you want to collect evaluation of faucet parameters in one place, then add checking that the faucet URL is not empty
like
```
if use faucet and faucet url is not empty...

else

```

File: hardhat.config.js [hunk 4]
```diff
@@ -86,27 +86,54 @@ for (const f of fs.readdirSync(path.join(__dirname, 'hardhat'))) {
...
   }
-  console.log(`Wallet = ${data.wallet}, amount = ${data.amount}`);
-  try {
-    const result = await axios.post(Config.faucetUrl, data);
-  } catch (err) {
-    console.log(`Failed to send request to faucet: ${err}`);
+  console.log(`Request '${data.amount}' tokens for wallet ${data.wallet} from faucet`);
+
+  let success = false;
+  for (let i=0; i < 10; i++){
```
Comment (apetrovskiy) [inline hardhat.config.js]: It'll stop working immediately in environments like DevNet, where the DDoS protection is set. Those environments require one minute's timeout between faucet calls

File: hardhat.config.js (hunk 1 above)
Comment (gigimon) [reply hardhat.config.js:160]: yes, but if web3.eth.getBalance failed, all our process failed too

File: hardhat.config.js (hunk 2 above)
Comment (gigimon) [reply hardhat.config.js:114]: It's used in function "getFaucetAndBalance". Why 10 times? If first call will success it's will last.

File: hardhat.config.js (hunk 4 above)
Comment (gigimon) [reply hardhat.config.js]: Which ddos protection? If first call success - no any more additional calls. If we get network problem (we sometimes get it) it reconnect us.

**Comment by apetrovskiy** [comment]
10:16 is here
image

**Comment by apetrovskiy** [comment]
[It's almost 10k LOC before the first test](https://github.com/neonlabsorg/neon-compatibility/runs/4642641168?check_suite_focus=true#step:9:9723)

File: hardhat.config.js [hunk 5]
```diff
@@ -86,27 +86,38 @@ for (const f of fs.readdirSync(path.join(__dirname, 'hardhat'))) {
...
-    const result = await axios.post(Config.faucetUrl, data);
-  } catch (err) {
-    console.log(`Failed to send request to faucet: ${err}`);
-  }
+  console.log(`Request '${data.amount}' tokens for wallet ${data.wallet} from faucet`);
+
+    try {
+      await axios.post(Config.faucetUrl, data);
+    } catch (err) {
+      console.log(`Failed to send request to faucet: ${err.response.data}`);
```
Comment (apetrovskiy) [inline hardhat.config.js]: It makes sense to add status code too:
`err.response.status`

**Review by apetrovskiy — DISMISSED** [review]

File: hardhat.config.js [hunk 6]
```diff
@@ -86,27 +86,38 @@ for (const f of fs.readdirSync(path.join(__dirname, 'hardhat'))) {
...
-  console.log(`Wallet = ${data.wallet}, amount = ${data.amount}`);
-  try {
-    const result = await axios.post(Config.faucetUrl, data);
-  } catch (err) {
-    console.log(`Failed to send request to faucet: ${err}`);
-  }
+  console.log(`Request '${data.amount}' tokens for wallet ${data.wallet} from faucet`);
+
+    try {
+      await axios.post(Config.faucetUrl, data);
```
Comment (apetrovskiy) [inline hardhat.config.js]: Ivan also asked for displaying (at least, or checking) status code like [here](https://github.com/axios/axios):
image

Comment (apetrovskiy) [reply hardhat.config.js]: `axios.post(x, y)`
`    .then((response) => {`
`        console.log(response.status);`
`        console.log(response.data); // not needed - nothing is displayed in the successful case`
`});`

Comment (apetrovskiy) [reply hardhat.config.js]: image

**Comment by gigimon** [comment]
@apetrovskiy comments fixed: https://github.com/neonlabsorg/neon-compatibility/runs/4666291040?check_suite_focus=true

**Review by apetrovskiy — APPROVED** [review]

**Pull request merged**